lib/spack/docs/developer_guide.rst

```diff
@@ -495,3 +495,217 @@ The bottom of the output shows the top most time consuming functions,
+   <https://github.com/spack/spack/commit/8eeb64096c98b8a43d1c587f13ece743c864fba9>`_.
+
+5. Update `CHANGELOG.md` with major highlights in bullet form. Use
+   proper markdown formatting, like `this example from 0.15.0
+   <https://github.com/spack/spack/commit/d4bf70d9882fcfe88507e9cb444331d7dd7ba71c>`_.
+
+6. Push the release branch to GitHub.
+
+7. Make sure CI passes on the release branch, including:
+   * Regular unit tests
+   * Build tests
+   * The E4S pipeline at `gitlab.spack.io <https://gitlab.spack.io>`_
+
+   If CI is not passing, submit pull requests to ``develop`` as normal
+   and keep rebasing the release branch on ``develop`` until CI passes.
+
+8. Go to `github.com/spack/spack/releases
+   <https://github.com/spack/spack/releases>`_ and click ``Draft a new
+   release``.  Set the following:
+
+   * ``Tag version`` should start with ``v`` and contain *all three*
+     parts of the version, e.g. ``v0.15.0``. This is the name of the tag
+     that will be created.
+
+   * ``Target`` should be the ``releases/vX.Y`` branch you just created.
+
+   * ``Release title`` should be ``vX.Y.Z`` (To match the tag, e.g. ``v0.15.0``).
+
+   * For the text, paste the latest release markdown from your ``CHANGELOG.md``.
+
+   You can save the draft and keep coming back to this as you prepare the release.
+
+9. When you are done, click ``Publish release``.
+
+10. Immediately after publishing, go back to
+    `github.com/spack/spack/releases
+    <https://github.com/spack/spack/releases>`_ and download the
+    auto-generated ``.tar.gz`` file for the release. It's the ``Source
+    code (tar.gz)`` link.
+
+11. Click “edit” on the release you just did and upload the downloaded
+    release tarball as a binary. This does two things:
+
+    1. Makes sure that the hash of our releases doesn’t change over time.
+       GitHub sometimes annoyingly changes they way they generate
+       tarballs, and then hashes can change if you rely on the
+       auto-generated tarball links.
+
+    2. Gets us download counts on releases visible through the GitHub
+       API. GitHub tracks downloads of artifacts, but *not* the source
+       links. See the `releases
+       page <https://api.github.com/repos/spack/spack/releases>`_ and search
+       for ``download_count`` to see this.
+
+12. Make a tweet like `this one
+    <https://twitter.com/spackpm/status/1231761858182307840>`_ to announce
+    the release, and tweet it out.
+
+13. Ping ``@channel`` in ``#general`` on `spackpm.slack.com
+    <https://spackpm.slack.com>`_ with a link to the tweet. The tweet
+    will be shown inline so that you do not have to retype your release
+    announcement.
+
+14. Email the Spack mailing list to let them know about the release.
+    Here's an `example email
+    <https://groups.google.com/forum/#!topic/spack/WT4CT9i_X4s>`_.
+
+15. Once all that is done, make sure that the release is merged into
```

Comment:
Why is this step not earlier in the process?

Reply:
I put it it slightly earlier in the latest version (and gave it its own section) -- see if you like it.